source/Sashimi.Azure.Accounts/AzureServicePrincipalAccountServiceMessageHandler.cs

```diff
@@ -10,8 +9,25 @@ namespace Sashimi.Azure.Accounts
     class AzureServicePrincipalAccountServiceMessageHandler : ICreateAccountDetailsServiceMessageHandler
     {
         public string AuditEntryDescription => "Azure Service Principal account";
-        public string ServiceMessageName => CreateAzureAccountServiceMessagePropertyNames.Name;
-        public IEnumerable<ScriptFunctionRegistration> ScriptFunctionRegistrations { get; } = Enumerable.Empty<ScriptFunctionRegistration>();
+        public string ServiceMessageName => CreateAzureAccountServiceMessagePropertyNames.CreateAccountName;
+        public IEnumerable<ScriptFunctionRegistration> ScriptFunctionRegistrations { get; } = new List<ScriptFunctionRegistration>
+        {
+            new ScriptFunctionRegistration("OctopusAzureServicePrincipalAccount",
+                                           "Creates a new Azure Service Principal Account.",
+                                           CreateAzureAccountServiceMessagePropertyNames.CreateAccountName,
+                                           new Dictionary<string, FunctionParameter>
+                                           {
+                                               { CreateAzureAccountServiceMessagePropertyNames.NameAttribute, new FunctionParameter(ParameterType.String) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.SubscriptionAttribute, new FunctionParameter(ParameterType.String) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.ApplicationAttribute, new FunctionParameter(ParameterType.String) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.TenantAttribute, new FunctionParameter(ParameterType.String) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.PasswordAttribute, new FunctionParameter(ParameterType.String) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.EnvironmentAttribute, new FunctionParameter(ParameterType.String, CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.EnvironmentAttribute) },
+                                               { CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.BaseUriAttribute, new FunctionParameter(ParameterType.String, CreateAzureAccountServiceMessagePropertyNames.ServicePrincipal.EnvironmentAttribute) },
```

Comment:
You missing https://github.com/OctopusDeploy/Calamari/pull/662/files#diff-a0b12e1993bb9ddb5c334109ee19cfc3L203, is that handled somewhere else now?

Reply:
No one is even using this `type` as you can probably see it in the `CreateAccountDetails` method.